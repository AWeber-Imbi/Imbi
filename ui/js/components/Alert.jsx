import React, { useEffect, useRef, useState } from 'react'

import PropTypes from 'prop-types'
import { Alert } from 'reactstrap'

export const propTypes = {
    children: PropTypes.array,
    color: PropTypes.string,
    noClose: PropTypes.bool,
    timeout: PropTypes.number,
}

const defaultTimeout = 10000

const icons = {
    danger: 'fas fa-exclamation-triangle',
    info: 'fas fa-info-circle',
    light: 'fas fa-exclamation-circle',
    warning: 'fas fa-exclamation-triangle',
    success: 'fas fa-info-circle',
}

export default function (props) {
    const timeoutID = useRef()

    const [display, setDisplay] = useState(false)

    useEffect(() => {
        setDisplay(props.children !== undefined)
    }, [props.children])

    function onToggle(e) {
        e.preventDefault()
        setDisplay(!display)
    }

    useEffect(() => {
        if (
            props.noClose !== true &&
            props.children !== undefined &&
            timeoutID.current === undefined
        ) {
            timeoutID.current = setTimeout(
                () => {
                    setDisplay(false)
                },
                props.timeout !== undefined ? props.timeout : defaultTimeout
            )
        }
        return () => {
            if (timeoutID.current !== undefined) clearTimeout(timeoutID.current)
        }
    }, [props.children, timeoutID])

    if (display === false) return null
    return (
        <Alert
            color={props.color}
            isOpen={true}
            fade={false}
            toggle={props.noClose === true ? null : onToggle}
        >
            <span
                className={
                    props.icon !== undefined ? props.icon : icons[props.color]
                }
            ></span>{' '}
            {props.children}
        </Alert>
    )
}
