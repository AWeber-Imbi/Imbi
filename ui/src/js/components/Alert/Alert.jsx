import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {
  faExclamationCircle, faInfoCircle, faExclamationTriangle,
  faCheckCircle
} from '@fortawesome/free-solid-svg-icons'
import PropTypes from "prop-types";
import React from 'react'

const icons = {
  info: faInfoCircle,
  warning: faExclamationTriangle,
  error: faExclamationCircle,
  success: faCheckCircle
}

function Alert({level, children}) {
  return (
    <div className={'alert-' + level}>
      <div className="flex">
        <div className="flex-shrink-0">
          <FontAwesomeIcon icon={icons[level]}/>
        </div>
        <div className="ml-3">
          {typeof children == 'string' ? (<h3 className="font-medium">{children}</h3>) : children}
        </div>
      </div>
    </div>
  )
}

Alert.propTypes = {
  level: PropTypes.oneOf(['info', 'warning', 'error', 'success']).isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
}

export default Alert
