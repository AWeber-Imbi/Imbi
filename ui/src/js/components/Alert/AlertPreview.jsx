import React from 'react'

import { ContentArea } from '../'

import { Alert } from './Alert'

class AlertPreview extends React.PureComponent {
  render() {
    return (
      <ContentArea
        pageTitle="Alerts"
        pageIcon="fas camera"
        className="space-y-5 w-full">
        <Alert level="error">This is an ERROR level alert</Alert>
        <Alert level="info">This is an INFO level alert</Alert>
        <Alert level="success">This is an SUCCESS level alert</Alert>
        <Alert level="warning">This is an WARNING level alert</Alert>
      </ContentArea>
    )
  }
}
export { AlertPreview }
