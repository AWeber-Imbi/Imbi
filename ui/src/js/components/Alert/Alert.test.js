import React from "react"
import {render} from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect"

import Alert from "./Alert"

describe("Alert", () => {
  it("should render an alert with info attributes", () => {
    const tree = render(<Alert level="info">Info</Alert>)
    expect(tree).toMatchSnapshot()
  })
  it("should render an alert with warning attributes", () => {
    const tree = render(<Alert level="warning">Warning</Alert>)
    expect(tree).toMatchSnapshot()
  })
  it("should render an alert with error attributes", () => {
    const tree = render(<Alert level="error">Error</Alert>)
    expect(tree).toMatchSnapshot()
  })
  it("should render an alert with success attributes", () => {
    const tree = render(<Alert level="success">Success</Alert>)
    expect(tree).toMatchSnapshot()
  })
  it("should render an alert with an object for a child", () => {
    const tree = render(<Alert level="info"><span>Foo</span></Alert>)
    expect(tree).toMatchSnapshot()
  })
})
