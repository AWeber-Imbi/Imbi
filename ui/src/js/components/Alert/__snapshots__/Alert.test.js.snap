// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render an alert with an object for a child 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert-info"
      >
        <div
          class="flex"
        >
          <div
            class="flex-shrink-0"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-info-circle fa-w-16 "
              data-icon="info-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="ml-3"
          >
            <span>
              Foo
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert-info"
    >
      <div
        class="flex"
      >
        <div
          class="flex-shrink-0"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-info-circle fa-w-16 "
            data-icon="info-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="ml-3"
        >
          <span>
            Foo
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Alert should render an alert with error attributes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert-error"
      >
        <div
          class="flex"
        >
          <div
            class="flex-shrink-0"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-circle fa-w-16 "
              data-icon="exclamation-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="ml-3"
          >
            <h3
              class="font-medium"
            >
              Error
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert-error"
    >
      <div
        class="flex"
      >
        <div
          class="flex-shrink-0"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-circle fa-w-16 "
            data-icon="exclamation-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="ml-3"
        >
          <h3
            class="font-medium"
          >
            Error
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Alert should render an alert with info attributes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert-info"
      >
        <div
          class="flex"
        >
          <div
            class="flex-shrink-0"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-info-circle fa-w-16 "
              data-icon="info-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="ml-3"
          >
            <h3
              class="font-medium"
            >
              Info
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert-info"
    >
      <div
        class="flex"
      >
        <div
          class="flex-shrink-0"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-info-circle fa-w-16 "
            data-icon="info-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="ml-3"
        >
          <h3
            class="font-medium"
          >
            Info
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Alert should render an alert with success attributes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert-success"
      >
        <div
          class="flex"
        >
          <div
            class="flex-shrink-0"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-check-circle fa-w-16 "
              data-icon="check-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="ml-3"
          >
            <h3
              class="font-medium"
            >
              Success
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert-success"
    >
      <div
        class="flex"
      >
        <div
          class="flex-shrink-0"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-check-circle fa-w-16 "
            data-icon="check-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="ml-3"
        >
          <h3
            class="font-medium"
          >
            Success
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Alert should render an alert with warning attributes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="alert-warning"
      >
        <div
          class="flex"
        >
          <div
            class="flex-shrink-0"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
              data-icon="exclamation-triangle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="ml-3"
          >
            <h3
              class="font-medium"
            >
              Warning
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="alert-warning"
    >
      <div
        class="flex"
      >
        <div
          class="flex-shrink-0"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
            data-icon="exclamation-triangle"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="ml-3"
        >
          <h3
            class="font-medium"
          >
            Warning
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
